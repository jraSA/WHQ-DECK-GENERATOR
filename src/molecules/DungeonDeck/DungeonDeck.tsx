import React, { useState } from 'react';
import { View } from 'react-native';
import styled from 'styled-components/native';

import dungeonDeck, { back, objectiveInstances } from '../../assets/dungeonDeck';
import Card from '../../atoms/Card';
import shuffle from '../../helpers/shuffle/Shuffle';
import Deck from '../Deck';

const DeckContainer = styled.View`
    flex: 1;
    flex-direction: row;
    align-self: center;
`;

const DungeonDeck = () => {
    const getRandomInstance = () => Math.floor(Math.random() * objectiveInstances.length);

    const [objectiveInstanceIndex, setObjectiveInstanceIndex] = useState(getRandomInstance());

    const getObjectiveInstanceIndexInsideDeck = (draft: any[]) => draft.length - 6 + Math.floor(Math.random() * 6);

    const generateDungeonDeck = () => {
        const draft = shuffle(dungeonDeck).slice(0, 11);
        const index = getObjectiveInstanceIndexInsideDeck(draft);

        draft.splice(index, 0, objectiveInstances[objectiveInstanceIndex]);
        return draft;
    };
    const [initialDeck, setInitialDeck] = useState(generateDungeonDeck());

    const resetDungeonDeck = () => {
        setObjectiveInstanceIndex(getRandomInstance());
        setInitialDeck(generateDungeonDeck());
    };

    return (
        <DeckContainer>
            <View>
                <Card source={objectiveInstances[objectiveInstanceIndex]} onRevert={resetDungeonDeck} />
            </View>
            <Deck generateDeck={generateDungeonDeck} back={back} initialDeck={initialDeck} />
        </DeckContainer>
    );
};

export default DungeonDeck;
