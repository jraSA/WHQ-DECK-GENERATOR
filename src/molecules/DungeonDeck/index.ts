import DungeonDeck from './DungeonDeck';

export default DungeonDeck;
