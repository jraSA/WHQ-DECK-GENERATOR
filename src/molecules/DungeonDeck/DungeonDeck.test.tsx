import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import DungeonDeck from './index';

describe('EventsTable Snapshot test', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<DungeonDeck />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
