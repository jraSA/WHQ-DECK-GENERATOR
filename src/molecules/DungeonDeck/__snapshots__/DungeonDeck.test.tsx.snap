// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsTable Snapshot test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": 233,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 22,
              "height": 360,
              "width": 233,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "borderRadius": 22,
                "height": 360,
                "width": 233,
              },
              null,
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00609e",
                "borderRadius": 24,
                "height": 44,
                "justifyContent": "center",
                "margin": 7,
                "width": 44,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": 233,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 22,
              "height": 360,
              "width": 233,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "borderRadius": 22,
                "height": 360,
                "width": 233,
              },
              null,
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00609e",
                "borderRadius": 24,
                "height": 44,
                "justifyContent": "center",
                "margin": 7,
                "width": 44,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 360,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": 233,
          },
        ]
      }
    />
  </View>
</View>
`;
